Source/WebCore/html/HTMLVideoElement.cpp

```diff
@@ -355,7 +364,7 @@ void HTMLVideoElement::paintCurrentFrameInContext(GraphicsContext& context, cons
         player->setVisibleForCanvas(true); // Make player visible or it won't draw.
         visibilityStateChanged();
     }
-    context.paintFrameForMedia(*player, destRect);
+    player->playerPrivate()->paintCurrentFrameInContext(context, destRect);
```

Comment:
It would not hurt to introduce a MediaPlayer:: paintCurrentFrameInContext.
Ditto for the double sync IPC, right?